service/src/main/kotlin/fi/espoo/evaka/invoicing/service/FinanceDecisionGenerator.kt

```diff
@@ -51,6 +54,45 @@ class FinanceDecisionGenerator(
     final val v2 = env.feeDecisionGeneratorV2Enabled
     final val useV2Tables = v1 && v2
 
+    fun scheduleBatchGeneration(tx: Database.Transaction, clock: EvakaClock) {
+        if (!v2) return
+
+        val inserted =
+            tx.createUpdate(
+                    """
+WITH ids AS (
+    SELECT head_of_child AS head_of_family_id
+    FROM fridge_child
+
+    UNION
+
+    SELECT head_of_family_id
+    FROM fee_decision
+)
+INSERT INTO async_job(type, payload, retry_count, retry_interval)
```

Comment:
This will break if the GenerateFinanceDecisions async job is ever changed. I think it would be better to plan with the Kotlin API *but* in this case we're talking about a very large amount of rows and AsyncJobRunner doesn't use batch inserts internally so it would come with a performance cost :shrug: